Tests/CommonTests/AssemblyResolverTests.cs

```diff
@@ -154,18 +167,20 @@ public void AssemblyResolver_ResolutionByFileNameWithSpace_Succeeds()
         [TestMethod]
         public void AssemblyResolver_VersionAssemblyRequested()
         {
-            // Arrange
+            // Setup
             string testFolder = TestUtils.CreateTestDirectory(this.TestContext);
             Assembly testAssembly = CompileSimpleAssembly("VersionAsm1.dll", testFolder, new TestLogger(), "2.1.0.4");
 
-            // Act
+            // 1. Search for a version that can be found -> succeeds
             Assembly resolvedAssembly = AssertAssemblyLoadSucceedsOnlyWithResolver("VersionAsm1, Version = 2.1.0.4, Culture = neutral, PublicKeyToken = null", testFolder);
-
-            // Assert
             AssertExpectedAssemblyLoaded(testAssembly, resolvedAssembly);
 
-            AssertAssemblyLoadFails("VersionAsm1, Version = 1.0.0.4, Culture = neutral, PublicKeyToken = null");
```

Comment:
This check wasn't exercising the product code so it didn't catch the product error.